src/frontend/js/modules/sidebar.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Local storage key
+ */
+const LOCAL_STORAGE_KEY = 'docs_siedabar_state';
+
+/**
+ * Sidebar module
+ */
+export default class Sidebar {
+  /**
+   * CSS classes
+   *
+   * @returns {Record<string, string>}
+   */
+  static get CSS() {
+    return {
+      toggler: 'docs-sidebar__section-toggler',
+      togglerCollapsed: 'docs-sidebar__section-toggler--collapsed',
+      section: 'docs-sidebar__section',
+      sectionAnimated: 'docs-sidebar__section--animated',
+      sectionTitle: 'docs-sidebar__section-title',
+      sectionTitleActive: 'docs-sidebar__section-title--active',
+      sectionList: 'docs-sidebar__section-list',
+      sectionListCollapsed: 'docs-sidebar__section-list--collapsed',
+      sectionListItemActive: 'docs-sidebar__section-list-item--active',
+      sidebarToggler: 'docs-sidebar__toggler',
+      sidebar: 'docs-sidebar',
+      sidebarHidden: 'docs-sidebar--hidden',
+    };
+  }
+
+  /**
+   * Creates base properties
+   */
+  constructor() {
+    this.nodes = {};
+    const storedState = window.localStorage.getItem(LOCAL_STORAGE_KEY);
+
+    this.collapsed = storedState ? JSON.parse(storedState) : {};
+  }
+
+  /**
+   * Called by ModuleDispatcher to initialize module from DOM
+   *
+   * @param {writingSettings} settings - module settings
+   * @param {HTMLElement} moduleEl - module element
+   */
+  init(settings, moduleEl) {
+    this.nodes.sections = Array.from(moduleEl.querySelectorAll('.' + Sidebar.CSS.section));
+    this.nodes.sections.forEach(section => {
+      const id = section.getAttribute('data-id');
+      const togglerEl = section.querySelector('.' + Sidebar.CSS.toggler);
+
+      if (!togglerEl) {
+        return;
+      }
+
+      togglerEl.addEventListener('click', e => this.handleSectionTogglerClick(id, section, togglerEl, e));
+
+      if (typeof this.collapsed[id] === 'undefined') {
+        this.collapsed[id] = false;
+      }
+      if (this.collapsed[id]) {
+        this.setSectionCollapsed(section, togglerEl, true, false);
+      }
+    });
+    this.nodes.sidebar = moduleEl.querySelector('.' + Sidebar.CSS.sidebar);
+    this.nodes.toggler = moduleEl.querySelector('.' + Sidebar.CSS.sidebarToggler);
+    this.nodes.toggler.addEventListener('click', () => this.toggleSidebar());
+    this.ready();
+  }
+
+  /**
+   * Toggles section expansion
+   *
+   * @param {number} sectionId - id of the section to toggle
+   * @param {HTMLElement} sectionEl - section html element
+   * @param {HTMLElement} togglerEl - toggler button html element
+   * @param {MouseEvent} event - click event
+   * @returns {void}
+   */
+  handleSectionTogglerClick(sectionId, sectionEl, togglerEl, event) {
+    event.preventDefault();
+    this.collapsed[sectionId] = !this.collapsed[sectionId];
+    window.localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(this.collapsed));
```

Comment:
pls, move the methods for working with the storage to the separate utility class, 

like this (rough example): 

```js
class Storage {
  constructor(prefix){
  }

  setItem(name, value){}
  
  getItem(name){}
}

class SidebarStorage extends Storage {
  constructor(){
    super('sidebar-state')

    this.keyName = 'sections-collapsed'
  }

  set(value){
    super.setItem(this.keyName, value)
  }

  get(){
    return super.getItem(this.keyName)
  }
}
```